### ISSUE TITLE
Scrolling does not work in the embedded application on mobile

### ISSUE THREAD

**Issue by neuvyaz0k** [flutter/flutter#138985]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Run app from this repo https://github.com/QQOld/embedded_app

### Expected results

Normal scrolling in embedded app on mobile

### Actual results

Scroll on mobile doesn't work (watch the attached video)

**Details**
At the moment I'm using iframe to embed flutter applications on the web. Its (iframe) problem is that there is no way (as far as I know) to link flutter and browser scrolling events. That is, scrolling the flutter app itself works fine, but we can't scroll a page with a flutter iframe by hovering over (or sliding over) the embedded flutter app. This is a rather kritical problem in my case.

After the advent of embedding a solution to this problem became possible, it seemed to me. I was able to handle flutter scroll events and scroll the browser if necessary. But then I noticed that scrolling on mobile is broken in principle, which makes embedding useless for me.

### Code sample

[Reproduction on GitHub](https://github.com/QQOld/embedded_app)

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

https://github.com/flutter/flutter/assets/71334848/f3ca8046-4874-48ec-80f3-c65db6931ebc

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.16.0, on Ubuntu 22.04.3 LTS 6.2.0-37-generic, locale ru_RU.UTF-8)
    • Flutter version 3.16.0 on channel stable at /home/qqold/sdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db7ef5bf9f (9 дней назад), 2023-11-15 11:25:44 -0800
    • Engine revision 74d16627b9
    • Dart version 3.2.0
    • DevTools version 2.28.2

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /home/qqold/Android/Sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /home/qqold/.local/share/JetBrains/Toolbox/apps/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • Ubuntu clang version 14.0.0-1ubuntu1.1
    • cmake version 3.22.1
    • ninja version 1.10.1
    • pkg-config version 0.29.2

[✓] Android Studio (version 2022.3)
    • Android Studio at /home/qqold/.local/share/JetBrains/Toolbox/apps/android-studio
    • Flutter plugin version 76.3.2
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] IntelliJ IDEA Community Edition (version 2023.2)
    • IntelliJ at /home/qqold/.local/share/JetBrains/Toolbox/apps/intellij-idea-community-edition
    • Flutter plugin version 75.1.4
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.84.2)
    • VS Code at /usr/share/code
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 22.04.3 LTS 6.2.0-37-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 119.0.6045.159

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by huycozy** [comment]
Hi @QQOld 
>Normal scrolling in embedded app on mobile

Could you be more specific on this? If you deploy web and open it on mobile browser, the scrollbar will not appear on Flutter view. So I'm not sure what the correct behavior should be.

**Comment by QQOld** [comment]
Hi, huycozy.
>If you deploy web and open it on mobile browser, the scrollbar will not appear on Flutter view.

I'm not quite sure what you meant here.
What I'm saying is that if we use as an embedded application an application that is fully vertically scrollable (as in the example above), then we can't scroll our embedded flutter app (like in the video).
In the video you can see that I'm trying to scroll the embedded flutter app, and the main browser window scrolls (while the scrolling in the embedded app also responds and shifts a bit)
The Iframe, which is essentially an alternative to the embedded app, scrolls normally in this case.

**Comment by huycozy** [comment]
I thought you mentioned the scrollbar on the embedded Flutter app is not draggable well in [Simulate mobile devices with Device Mode](https://developer.chrome.com/docs/devtools/device-mode/). What I meant is that scrollbar will not appear in the "real" case when we deploy the web and open it on mobile browser, like this:

<details>
<summary>Demo (web on mobile browser)</summary>

https://github.com/flutter/flutter/assets/104349824/d2c33284-b2c6-424c-a51d-2db670cbb411
</details>

But it is not what the issue was about. 

>What I'm saying is that if we use as an embedded application an application that is fully vertically scrollable (as in the example above), then we can't scroll our embedded flutter app

Thanks, it's more clear. But I wonder what the expected scroll behavior should be. When the touch point is in the embedded Flutter app, should it prioritize the scroll list there instead of the parent view (main browser window)?

May be related: https://github.com/flutter/flutter/issues/78216

**Comment by QQOld** [comment]
Yes, I have seen this issue. The issues may be partially related, but as I wrote above, in the iframe in the identical case, scrolling works (example below). And we're talking about a new embedding, not an iframe, so I think that's a separate issue.
<details><summary>Example:</summary>

https://github.com/flutter/flutter/assets/71334848/a228f4ab-02df-4629-bbb7-ba5a6f450f66

</details> 

And the [issue](https://github.com/flutter/flutter/issues/78216) you referenced is a pretty big problem for embedding with iframes, since, for example on a mobile device browser, with a fairly large iframe size, it can become impossible to scroll through the main content of the browser page.
In the [documentation](https://docs.flutter.dev/deployment/web#embedding-a-flutter-app-into-an-html-page), flutter embedding and iframe are in the same section, which is also why I expected the same scroll behavior in iframe and embedding.

I think I was able to solve the problem described in the [issue](https://github.com/flutter/flutter/issues/78216) by replacing the iframe with an embedded app (for now at least when scrolling with the mouse wheel, but I don't think there would be a problem with mobile either). My point is that if touch point will give priority to the list inside flutter app, then we can handle flutter scrolling events, and when we reach the end of the embedded scrolling we can start scrolling the main browser window with js (and that's not possible with the current behavior). Also, this behavior would be similar to iframe, which is the original way of embedding (but without the ability to interact with js).

**Comment by huycozy** [comment]
Thank you for the detailed explanation! Reproduced this issue on master channel as well.

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.16.1, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.16.1 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7f20e5d18c (9 hours ago), 2023-11-27 09:47:30 -0800
    • Engine revision 22b600f240
    • Dart version 3.2.1
    • DevTools version 2.28.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.8+0-17.0.8b1000.22-10799086)

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe Patch 3.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.76.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 119.0.6045.159

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

```bash
[!] Flutter (Channel master, 3.17.0-17.0.pre.47, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.17.0-17.0.pre.47 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7a0508e5de (10 minutes ago), 2023-11-28 22:01:04 -0500
    • Engine revision 4beaa1195b
    • Dart version 3.3.0 (build 3.3.0-170.0.dev)
    • DevTools version 2.30.0-dev.4
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.8+0-17.0.8b1000.22-10799086)

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe Patch 3.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.76.0

[✓] Connected device (4 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D                         • android-arm64  • Android 11 (API 30)
    • iPhone (mobile)  • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.8 19H370
    • macOS (desktop)  • macos                                    • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome                                   • web-javascript • Google Chrome 119.0.6045.159

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```
</details>

**Comment by yjbanov** [comment]
This is a duplicate of https://github.com/flutter/flutter/issues/139263

**Comment by ditman** [comment]
This shouldn't have been deduplicated into https://github.com/flutter/flutter/issues/139263, this is an actual issue when attempting to scroll flutter content, embedded, in a device capable of "tap" (a phone or the browser tap emulator). I've reproduced this both in an iPhone, and Chrome and Firefox with "Tap" support enabled.

I've worked on a fix, and am reopening this issue so we can link that fix to this issue.
